test/smoke/kogitoapp.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+)
+
+const (
+	// Kiegroup kogito-examples quarkus examples are missing some properties to handle correctly the persistence with Infinispan.
+	// Quarkus 1.1.0 integration should correct that.
+	//kogitoExamplesURI = "https://github.com/kiegroup/kogito-examples"
+	kogitoExamplesURI = "https://github.com/radtriste/kogito-examples.git"
+)
+
+// DeployQuarkusExample deploy a Quarkus example
+func DeployQuarkusExample(namespace, appName, contextDir string, native, persistence bool) error {
+	GetLogger(namespace).Infof("Deploy quarkus example %s with name %s, native %v and persistence %v", contextDir, appName, native, persistence)
+	return DeployExample(namespace, appName, contextDir, "quarkus", native, persistence)
+}
+
+// DeploySpringBootExample deploys a Spring boot example
+func DeploySpringBootExample(namespace, appName, contextDir string, persistence bool) error {
+	GetLogger(namespace).Infof("Deploy spring boot example %s with name %s and persistence %v", contextDir, appName, persistence)
+	return DeployExample(namespace, appName, contextDir, "springboot", false, persistence)
+}
+
+// DeployExample deploys an example
+func DeployExample(namespace, appName, contextDir, runtime string, native, persistence bool) error {
+	gitProjectURI := kogitoExamplesURI
+
+	kogitoApp := getKogitoAppStub(namespace, appName)
+	if runtime == "quarkus" {
+		kogitoApp.Spec.Runtime = v1alpha1.QuarkusRuntimeType
+	} else if runtime == "springboot" {
+		kogitoApp.Spec.Runtime = v1alpha1.SpringbootRuntimeType
+	}
+	kogitoApp.Spec.Build.Native = native
+	kogitoApp.Spec.Build.GitSource.URI = &gitProjectURI
+	kogitoApp.Spec.Build.GitSource.ContextDir = contextDir
+	kogitoApp.Spec.Build.GitSource.Reference = "0.6.1"
```

Comment:
Why there is used specific branch and not master?
If specific branch needs to be used, can it be configurable? For example using env variable?

Reply:
Branch master had problems was not compiling ... :S
I will set this one and the URI as configurable via env variable.